s3_transport/include/irods/private/s3_transport/s3_transport.hpp

```diff
@@ -1478,14 +1478,20 @@ namespace irods::experimental::io::s3_transport
                         // turn off trunc flag
                         std::ios_base::openmode mode;
                         bool trunc_flag = false;
-                        if (data.threads_remaining_to_close == 0) {
-                            trunc_flag = true;
+
+						// If we know the number of threads, use threads_remaining_to_close to determine if this is the
+						// first open. If we do not know the number of threads, use file_open_counter.
+						if ((data.know_number_of_threads && 0 == data.threads_remaining_to_close) ||
+						    (!data.know_number_of_threads && 0 == data.file_open_counter))
+						{
+							trunc_flag = true;
                             mode = mode_;
-                        } else {
-                            mode = mode_ & ~std::ios_base::trunc;
-                        }
+						}
+						else {
+							mode = mode_ & ~std::ios_base::trunc;
+						}
 
-                        // try opening for read and write, if it fails create then open for read/write
+						// try opening for read and write, if it fails create then open for read/write
```

Comment:
Consider tweaking the wording around the use of "create". I think it might confuse future readers.
```
..., if it fails create then open for read/write
```

Reply:
I clarified the wording a bit.